### ISSUE TITLE
Extrusions don't work on mobile safari on iPhone 10 running iOS 11.4.1

### ISSUE THREAD

**Issue by brsbl** [mapbox/mapbox-gl-js#7104]
**mapbox-gl-js version**: v0.47.0

**browser**: mobile safari on iPhone 10

### Steps to Trigger Behavior

 1. Open a map with extrusions on mobile safari on an iPhone 10

### Link to Demonstration

Here's an example url: https://api.mapbox.com/styles/v1/saman/cjkn8e4ki85r62qmw46cqgy2k.html?fresh=true&title=true&access_token=pk.eyJ1Ijoic2FtYW4iLCJhIjoiS1ptdnd0VSJ9.19qza-F_vXkgpnh80oZJww#0.0/0.000000/0.000000/0

And a picture of the linked style working on iPhone 8 and not iPhone 10 (both running iOS 11.4.1):
![image from ios](https://user-images.githubusercontent.com/19296229/43932487-45eb077a-9bf9-11e8-9332-7562b892666c.jpg)

Also found it worked on iPhone 7.

Style sheet for this style:
```
{
    "version": 8,
    "name": "Blank",
    "center": [
        1.4622770665675944,
        42.27278859987004
    ],
    "zoom": 5.14961062357403,
    "bearing": -20.000000000000114,
    "pitch": 36.50000000000002,
    "sources": {
        "composite": {
            "url": "mapbox://mapbox.mapbox-streets-v7",
            "type": "vector"
        }
    },
    "sprite": "mapbox://sprites/saman/cjkn8e4ki85r62qmw46cqgy2k",
    "glyphs": "mapbox://fonts/saman/{fontstack}/{range}.pbf",
    "layers": [
        {
            "id": "background",
            "type": "background",
            "paint": {
                "background-color": "hsl(237, 76%, 64%)"
            }
        },
        {
            "id": "country-label",
            "type": "fill-extrusion",
            "source": "composite",
            "source-layer": "water",
            "layout": {},
            "paint": {
                "fill-extrusion-height": 100000,
                "fill-extrusion-color": "hsl(0, 0%, 100%)"
            }
        }
    ],
    "created": "2018-08-10T00:06:23.155Z",
    "id": "cjkn8e4ki85r62qmw46cqgy2k",
    "modified": "2018-08-10T00:09:34.783Z",
    "visibility": "private"
}
```

### Expected Behavior

Extrusions to appear

### Actual Behavior

Extrusions don't appear, just the background color of the map

**Comment by jfirebaugh** [comment]
Can someone with an iPhone 10 test https://github.com/mapbox/mapbox-gl-js/compare/extrusion-mediump?

**Comment by ryanhamley** [comment]
I'm still just seeing the background color

**Comment by ChrisLoer** [comment]
This should be fixed with #7292 -- I made that PR not seeing that John had done almost the same thing at https://github.com/mapbox/mapbox-gl-js/compare/extrusion-mediump. The only difference in my PR is that I went all the way to `highp` precision, following discussion from #2096. I was able to reproduce this on my iPhone 8 and the problem went away after the change.
